### ISSUE TITLE
Business BCeID Keycloak Realm

### ISSUE THREAD

**Issue by jakemorr** [bcgov/lucy-web#385]
**See Comments** 

**Describe the task**
the response with IDIM about BeID are becoming quite technical. 
**Acceptance Criteria**
- [ ] determine how we plan 
- [ ] second
- [ ] third

**Additional context**
- email from IDIM:
Let me rephrase the question a different way…
 
How does the InvasivesBC application know it is dealing with a Business BCeID?
What is the InvasivesBC application behaviour if a Basic BCeID or Personal BCeID comes knocking on your front door instead of a Business BCeID?
What BCeID account identity attribute(s) are you using to differentiate between specific BCeID accounts?
 
If your InvasivesBC application doesn’t have a good way to filter out Basic BCeID & Business BCeID accounts, then your draft BCeID use cases should be expanded to include Basic BCeID & Personal BCeID use cases.
 
Have you tested the non-production InvasivesBC application workflow using a Basic BCeID & a Personal BCeID?
 
If you need Personal BCeID & Business BCeID test accounts in the BCeID Client Test environment, please refer to these non-production registration guides:
https://citz.sp.gov.bc.ca/sites/SBC/IDIM/FG/Shared%20Documents/How to get a Personal BCeID account in BCeID non production environments.docx
https://citz.sp.gov.bc.ca/sites/SBC/IDIM/FG/Shared%20Documents/How to request accounts for Business BCeID client test and BCeID client development environments.docx
 
Some other outstanding questions:
 
Can you provide an update on your target go-live Production launch date? – [Obviously, we’ve blown past March 2020…]
 
Who on the InvasivesBC side holds the pen for your project’s communications / invitations (that mention BCeID)?
 
Your communications / invitations (that mention BCeID) must be reviewed and signed off by the IDIM Program before InvasivesBC will be permitted to integrate with our BCeID Production environment.
 
When you have questions or concerns, please contact us.

**Comment by jakemorr** [comment]
Good Afternoon Jake,
 
Thank you for letting us know the InvasivesBC application is currently ‘IDIR-only’ and that you are planning on integrating with Production BCeID in the Fall of 2020.
 
Strategies for how the InvasivesBC application could filter out Basic BCeID & Personal BCeID accounts from accessing the InvasivesBC application?
 
You could examine incoming BCeID accounts for the presence of the [ Business GUID ] identity attribute. [ Business GUID ( Businessidentifier ) is available via KeyCloak ]
Basic BCeID accounts & Personal BCeID accounts would not have a [ Business GUID ] associated with those account types…
 
I believe the EMBC ESS Project in the DevExchange Lab are leveraging KeyCloak for Business BCeID & examining accounts for the presence of the [ Business GUID ] is how they are handling the issue.
Rafael Ponce de Leon (Quartech) likely has the inside track on what you are looking to do.
 
There is a DevExchange project underway to segregate the KeyCloak Realms that are integrated with BCeID as follows:
 
[Business BCeID only]
[Basic BCeID only]
[Business & Basic BCeID only]
In the future, [Personal BCeID] will not be an option for KeyCloak integrated applications.
 
If you are interested in migrating the InvasivesBC application from the ‘universal’ KeyCloak Realm to either the [Business & Basic BCeID only] or [Business BCeID only] future realms, please contact Luke Gonis or Shelly Han for details and timelines.
 
Given your Fall 2020 Production BCeID integration timeline, migrating InvasivesBC to the [Business BCeID only] realm might be a good fit.
 
Best Regards,
 
Tim McGuire
Client Onboarding Manager - BCeID

**Comment by davidclaveau** [comment]
InvasivesBC has disabled BCeID login.
